getting-started/src/main/kotlin/com/google/cloud/kotlin/demo/EmojifyController.kt

```diff
@@ -0,0 +1,152 @@
+        }
+    }
+    return Emoji.NONE
+}
+
+class GcsObject(val bucketName: String, val blobName: String)
+class EmojifyResponse(val original: GcsObject, val emojified: GcsObject, val emojifiedUrl: String)
+
+@RestController
+class EmojifyController {
+    @GetMapping("/emojify")
+    fun emojify(@RequestParam(value = "objectName") objectName: String): EmojifyResponse? {
+
+        val publicUrl: String =
+            "https://storage.googleapis.com/${bucket.name}/emojified/emojified-$objectName" // api response
+        val log = Logger.getLogger(Application::class.java.name)
+
+        // Setting up image annotation request
+        val requests = ArrayList<AnnotateImageRequest>()
+        val vision = ImageAnnotatorClient.create()
+        val source = ImageSource.newBuilder().setGcsImageUri("gs://${bucket.name}/$objectName").build()
+        val img = Image.newBuilder().setSource(source).build()
+        val feat = Feature.newBuilder().setType(Type.FACE_DETECTION).build()
+        val request = AnnotateImageRequest.newBuilder()
+                .addFeatures(feat)
+                .setImage(img)
+                .build()
+
+        requests.add(request)
+
+        // Calls vision api on above image annotation requests
+        val response = vision.batchAnnotateImages(requests)
+        vision.close()
+
+        for (resp in response.responsesList) {
+            if (resp.hasError()) {
+                log.severe(resp.error.message)
+                return null
+            }
+            // Downloads source image to /tmp/<objectName>
+            val originalBlob = bucket.get(objectName)
+            val localFilePath = Paths.get("/tmp/$objectName")
+            val writeDest = PrintStream(FileOutputStream(localFilePath.toFile()))
+            writeDest.write(originalBlob.getContent())
+            writeDest.close()
+
+            val imgBuff = ImageIO.read(Paths.get("/tmp/$objectName").toFile())
+            val gfx = imgBuff.createGraphics()
+            val extension = (Paths.get("/tmp/$objectName").toFile().extension)
+
+            for (annotation in resp.faceAnnotationsList) {
+                val emoji = bestEmoji(annotation)
+                val imgEmoji = ImageIO.read(Paths.get(EMOJIS_PATH + emojiPic[emoji]).toFile())
+                log.info("EMOJI DETECTED: $emoji")
+                val poly = Polygon()
+                for (vertex in annotation.fdBoundingPoly.verticesList) {
+                    poly.addPoint(vertex.x, vertex.y)
+                }
+                val height = poly.ypoints[2] - poly.ypoints[0]
+                val width = poly.xpoints[1] - poly.xpoints[0]
+                // Draws emoji on detected face
+                gfx.drawImage(imgEmoji, poly.xpoints[0], poly.ypoints[1], height, width, null)
+            }
+            // Writing emojified image to temp file
+            val writeTo = "/tmp/emojify-$objectName"
+            ImageIO.write(imgBuff, extension, Paths.get(writeTo).toFile())
```

Comment:
i think it's unnecessary to write the image to a file and then reading it back again into bytes.